homework-submission/todo-app/src/todos/apiElements.jsx

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faCheck, faTrash, faPencilAlt, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { fab } from '@fortawesome/free-brands-svg-icons'
+import {inject, observer} from 'mobx-react';
+
+
+
+ 
+library.add(fab, faCheck, faTrash, faPencilAlt, faTimes);
+
+@inject('TodosStore')
+@observer
+class Element extends Component {
+    state = {
+        id: this.props.id,
+        description: this.props.description,
+        deadline: this.props.deadline,
+        done: this.props.done,
+        editMode: false,
+        temporarySave: this.props.description
+     }
+
+     throwLineHandle = () => {
+        (!this.state.done) ? this.setState({done: true}) : this.setState({done: false});
+        this.props.TodosStore.changeDoneValue(this.props.id);
+     }
+
+     turnUpdateMode = () => {
+        const temp = this.state.description;
+        this.setState({
+            editMode: !this.state.editMode,
+            temporarySave: temp
+        });
+     }
+
+     onChangeHandle = e => {
+         this.setState({[e.target.name]: e.target.value});
+     }
+
+     submitUpdate = e => {
+        e.preventDefault();
+        this.setState({editMode: !this.state.editMode});
+        const {id, description, deadline, done} = this.state;
+        const item = {
+            id: id,
```

Comment:
If you are updating an item, while you have an auto generated `id` then no need for inserting the id to it.
Even if you have to, the server is receiving `_id`. 🤔